backend/dataall/modules/notebooks/db/repositories.py

```diff
@@ -0,0 +1,53 @@
+"""
+DAO layer that encapsulates the logic and interaction with the database for notebooks
+Provides the API to retrieve / update / delete notebooks
+"""
+from sqlalchemy import or_
+from sqlalchemy.orm import Query
+
+from dataall.db import paginate
+from dataall.modules.notebooks.db.models import SagemakerNotebook
+
+
+class NotebookRepository:
+    """DAO layer for notebooks"""
+    _DEFAULT_PAGE = 1
+    _DEFAULT_PAGE_SIZE = 10
+
+    def __init__(self, session):
+        self._session = session
```

Comment:
not sure if session should be a part of the constructor, I think it should be rather obtainer on a request basic since it stores/tracks all the objects which are part of the session

Reply:
ok I raised it up since I am just get used to create all the stuff for the lambda function once before the handler method gets executed to reduce the cold start time and for the case the lambda env get reused for the next calls to not instantiate objects once again